README.md

```diff
@@ -20,9 +20,34 @@ Community-maintained package to support using [Rust](https://www.rust-lang.org/)
 
 The below documentation is for the `vercel_runtime` crate (in beta). If you are looking for the legacy runtime instructions using `vercel_lambda` see [tree/a9495a0](https://github.com/vercel-community/rust/tree/a9495a0f0d882a36ea165f1629fcc79c30bc3108).
 
+## Introduction
+
+This section aims to give you a basic understanding about Vercel runtimes.
+
+### Builder module
+
+The _npm_ module `vercel-rust` is implementing an interface which is primarily taking care of spawning a development server, caching between consecutive builds, and running the compilation. You can read more about the in-depths of implementing a builder [here](https://github.com/vercel/vercel/blob/main/DEVELOPING_A_RUNTIME.md).
+
+Note that this dependency **does not** have to be installed manually, as it is pulled automatically by the Vercel CLI or the build container during deployments.
+
+### Runtime crate
+
+The crate `vercel_runtime` is what you will consume in your Rust functions. As the name suggests, the runtime crate takes care of everything that happens during run-time. In specific it takes care of creating a [Tower](https://docs.rs/tower/latest/tower/trait.Service.html) service, which expects a specific handler signature. The flow of an invocation can be visualized as the following:
+
+```mermaid
+graph TD
+    A["Function Invocation"] --> |"process_request(event: InvocationEvent&lt;VercelEvent&gt;) → Request"| B[Request]
+    B --> |"handler_fn(req: Request) → Future&lt;Output = Result&lt;Response&lt;Body&gt;, Error&gt;&gt;"| C["Runtime calls handler_fn"]
+    C --> |"Ok(r) => process_response(r)"| D["Response"]
+```
+
 ## Usage
```

Comment:
Should be Getting Started, imo, rather than Usage. 

Needs to **clearly** show what steps need to be taken. Ideally numbered. 

e.g. 

1. Add a `vercel.json` file to your project: 
```
...
```

This turns every file matching `api/**/*.rs` into a Vercel Function. 

2. Create a function. As an example, here is `api/hello.rs`
```rust
...
```

3. Create a `Cargo.toml` (and Cargo.lock??). 
```toml
...
```
Note that each Vercel Function needs to be explicitly stated as a binary, and the `vercel_runtime` crate must be present. 

4. Add a `.vercelignore` to ignore build artefacts. 
```
...
```

Reply:
5. You're all set. Run `vercel dev` or `vercel` to build/deploy your thing.

Reply:
Q: should the create `Cargo.toml` step ask you to do: 
```bash
cargo init
```

Reply:
> Q: should the create `Cargo.toml` step ask you to do:
> 
> ```shell
> cargo init
> ```

This would either bootstrap a executable or library project + initialize a git repository. I feel like we'd then need to remove more than what we aim to create?